tests/conftest.py

```diff
@@ -30,6 +30,31 @@
 This also allows test to run a few hundred times before being marked as slow.
 """
 
+"""
+About venvs in tests (see linked objects' docstrings for more details):
+During normal operation the module loader and the compiler work with the following environments:
+- inmanta.env.process_env: inmanta.env.ActiveEnv -> presents an interface to interact with the outer (developer's) venv. Used by
+    inmanta.module.Project to install v2 modules in.
+- inmanta.module.Project.virtualenv: inmanta.env.VirtualEnv -> compiler venv. Used by the compiler to install v1 module
+    requirements in. Inherits from the outer venv.
+
+When running the tests we don't want to make changes to the outer environment. So for tests that install Python packages we need
+to make sure those are installed in a test environment. This means patching the inmanta.env.process_env to be an interface
+to a test environment instead of the outer environment.
+The following fixtures manage test environments:
+- snippetcompiler_clean: activates the Project's compiler venv and patches inmanta.env.process_env to use this same venv
+    as if it were the outer venv. The activation and patch is applied when compiling the first snippet.
+- snippetcompiler: same as snippetcompiler_clean but the compiler venv is shared among all functions using the
+    fixture.
+- tmpvenv: provides a decoupled test environment (does not inherit from the outer environment) but does not activate it and
```

Comment:
The docstring of the `tmpvenv` fixture mentions that it is activated.

Reply:
That's incorrect, thanks. I've updated the docstring.